From 1d958061eba677ec62b25746669b6553e98447c4 Mon Sep 17 00:00:00 2001
From: Shayne Chen <shayne.chen@mediatek.com>
Date: Mon, 18 Aug 2025 15:32:21 +0800
Subject: [PATCH 114/123] mtk: mt76: mt7996: add a debugFS to disable radar
 event reported

To disable radar event on extender mode for debug purpose:
hostapd_cli -i <interface_name> raw DFS_DETECT_MODE 1
echo 1 > /sys/kernel/debug/ieee80211/phy0/mt76/ignore_radar_event

Signed-off-by: Shayne Chen <shayne.chen@mediatek.com>
---
 mt7996/mcu.c         | 2 ++
 mt7996/mt7996.h      | 1 +
 mt7996/mtk_debugfs.c | 2 ++
 3 files changed, 5 insertions(+)

diff --git a/mt7996/mcu.c b/mt7996/mcu.c
index 990978c2..bf26900c 100644
--- a/mt7996/mcu.c
+++ b/mt7996/mcu.c
@@ -1648,6 +1648,8 @@ mt7996_mcu_uni_rx_unsolicited_event(struct mt7996_dev *dev, struct sk_buff *skb)
 		mt7996_mcu_ie_countdown(dev, skb);
 		break;
 	case MCU_UNI_EVENT_RDD_REPORT:
+		if (dev->dbg.ignore_radar_event)
+			break;
 		mt7996_mcu_rx_radar_detected(dev, skb);
 		break;
 	case MCU_UNI_EVENT_ALL_STA_INFO:
diff --git a/mt7996/mt7996.h b/mt7996/mt7996.h
index 70e7a5b2..27d168c9 100644
--- a/mt7996/mt7996.h
+++ b/mt7996/mt7996.h
@@ -986,6 +986,7 @@ struct mt7996_dev {
 		} lp;
 
 		bool beacon_loss_keep_conn;
+		bool ignore_radar_event;
 	} dbg;
 	const struct mt7996_dbg_reg_desc *dbg_reg;
 	bool sr_pp_enable:1;
diff --git a/mt7996/mtk_debugfs.c b/mt7996/mtk_debugfs.c
index b2b9a05c..9f4b449f 100644
--- a/mt7996/mtk_debugfs.c
+++ b/mt7996/mtk_debugfs.c
@@ -4763,6 +4763,8 @@ void mt7996_mtk_init_dev_debugfs(struct mt7996_dev *dev, struct dentry *dir)
 	debugfs_create_u8("sku_disable", 0600, dir, &dev->dbg.sku_disable);
 	debugfs_create_bool("beacon_loss_keep_conn", 0600, dir,
 			    &dev->dbg.beacon_loss_keep_conn);
+	debugfs_create_bool("ignore_radar_event", 0600, dir,
+			    &dev->dbg.ignore_radar_event);
 #endif
 
 	debugfs_create_file("muru_prot_thr", 0200, dir, dev, &fops_muru_prot_thr);
-- 
2.45.2

